### PR TITLE
es2015ify

### PR DESCRIPTION
**Pull request by SamVerschueren**
Too bad node 4 doesn't support spread. This would be so clean with that feature :).

### CODE REVIEW

**Comment by sarbbottam** [comment]
What could be the next module that needs to be `es2015fied`? I would like to give it a shot.

File: index.js [hunk 1]
```diff
@@ -1,7 +1,5 @@
 'use strict';
 module.exports = function (arr) {
-	var rest = [].concat.apply([], [].slice.call(arguments, 1));
-	return arr.filter(function (el) {
-		return rest.indexOf(el) === -1;
-	});
+	const rest = new Set([].concat.apply([], [].slice.call(arguments, 1)));
+	return arr.filter(el => !rest.has(el));
```
Comment (sindresorhus) [inline index.js]: `el` => `x`

File: index.js [hunk 2]
```diff
@@ -1,7 +1,5 @@
 'use strict';
 module.exports = function (arr) {
```
Comment (sindresorhus) [inline index.js]: arrow function

**Comment by sindresorhus** [comment]
@SamVerschueren Might as well wait on Node.js 6 which should be out in a week.

**Comment by sindresorhus** [comment]
@sarbbottam This one could be promisified and ES2015ified: https://github.com/sindresorhus/read-chunk/blob/master/index.js (You can use https://github.com/sindresorhus/pify to promisify the core `fs` APIs) ;)

File: index.js (hunk 2 above)
Comment (SamVerschueren) [inline index.js]: Arrow functions don't have `arguments` if I'm not mistaken.

**Comment by SamVerschueren** [comment]
@sindresorhus So wait for Node 6 and drop support for 4 then?

File: index.js (hunk 2 above)
Comment (sindresorhus) [inline index.js]: Oh right. Can be arrows when we change to use rest/spread though.

**Comment by sindresorhus** [comment]
Yes

**Comment by SamVerschueren** [comment]
It seems that even in Node 5.x.x it's behind a `--harmony_rest_parameters` flag. https://nodejs.org/en/docs/es6/#which-features-are-behind-the-es_staging-flag

**Comment by sarbbottam** [comment]
> > This one could be promisified and ES2015ified: https://github.com/sindresorhus/read-chunk/blob/master/index.js (You can use https://github.com/sindresorhus/pify to promisify the core fs APIs) ;)

Lol, its over!

**Comment by SamVerschueren** [comment]
@sindresorhus what are we going to do with this module? Only support Node 6 and use rest parameters?

**Comment by sindresorhus** [comment]
@SamVerschueren Yes

**Comment by SamVerschueren** [comment]
Implemented. This is so clean 😸 ! Feedback welcome!

**Comment by sindresorhus** [comment]
So beautiful ❤️

**Pull request merged**